test/streams-test.js

```diff
@@ -135,6 +135,12 @@ function checkDecodeStream(opts) {
 }
 
 describe("Streaming mode", function() {
+  	it("Encoding using internal modules: utf8 with surrogates in separate chunks", checkEncodeStream({
+        encoding: "utf8",
+        input: ["\uD83D", "\uDE3B"],
```

Comment:
Could you check that strings with more than one character are processed too? E.g. 
```
input: [
    "\uD83D", "\uDE3B", "\n", 
    "abc\uD83D", "\uDE3B", "\n", 
    "\uD83D", "\uDE3Befg", "\n", 
    "abc\uD83D", "\uDE3Befg", "\n", 
],
```

Also, non-matching surrogates at the boundary like low-low, high-high, high-low.

Also, make an input that starts with a low surrogate, but doesn't end with a high surrogate (`input: ["abc\uD83D"]`) - that should probably return the surrogate unchanged.

Reply:
I've thought of these questions, but not as comprehensively as you.
For example, I thought surrogates should be in pairs always.
In short, I completed some test cases and passed all of them.
It might not be perfect, but it's better.